frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch.ts

```diff
@@ -32,17 +32,30 @@ const consoleFetchCommon = async (
   method: string = 'GET',
   options: RequestInit = {},
   timeout?: number,
+  isFullResponse?: boolean,
 ) => {
   const headers = getConsoleRequestHeaders();
+  const defaultIsFullResponse = typeof isFullResponse !== 'boolean' ? false : isFullResponse;
+
   // Pass headers last to let callers to override Accept.
   const allOptions = _.defaultsDeep({ method }, options, { headers });
   const response = await consoleFetch(url, allOptions, timeout);
   const text = await response.text();
   const isPlainText = response.headers.get('Content-Type') === 'text/plain';
   if (!text) {
-    return isPlainText ? '' : {};
+    return isPlainText ? response : {};
   }
-  return isPlainText || !response.ok ? text : JSON.parse(text);
+
+  // Map cosole respose to data, headers and status
+  const fullResponse: { data: Body; headers: { [key: string]: any }; status: number } = {
+    headers: [...response.headers].reduce((acc, header) => {
+      return { ...acc, [header[0]]: header[1] };
+    }, {}),
+    status: response.status,
+    data: JSON.parse(text),
+  };
+
+  return !response.ok ? text : defaultIsFullResponse ? fullResponse : fullResponse.data;
```

Comment:
This doesn't seem right. Shouldn't we always return the full response when the flag is true? Also, generally we avoid nested tertiary conditions.
```suggestion
  const responseBody = !response.ok ? text : fullResponse.data
  return defaultIsFullResponse ? fullResponse : responseBody;
```